setup.py

```diff
@@ -6,73 +6,74 @@
 )
 
 extras_require = {
-    'test': [
-        "pytest>=6.2.5,<7",
-        "pytest-xdist",
-        "tox>=2.9.1,<3",
+    "test": [
+        "pytest>=7.0.0",
+        "pytest-xdist>=2.4.0",
     ],
-    'lint': [
-        "flake8==3.8.3",
-        "isort>=4.2.15,<5",
-        "mypy==0.910",
-        "pydocstyle>=3.0.0,<4",
+    "lint": [
+        "flake8==6.0.0",  # flake8 claims semver but adds new warnings at minor releases, leave it pinned.
+        "flake8-bugbear==23.3.23",  # flake8-bugbear does not follow semver, leave it pinned.
+        "isort>=5.10.1",
+        "mypy==0.971",  # mypy does not follow semver, leave it pinned.
+        "pydocstyle>=6.0.0",
+        "black>=22",
     ],
-    'doc': [
+    "doc": [
         "sphinx>=5.0.0",
         "sphinx_rtd_theme>=1.0.0",
         "towncrier>=21,<22",
     ],
-    'dev': [
-        "bumpversion>=0.5.3,<1",
-        "pytest-watch>=4.1.0,<5",
+    "dev": [
+        "bumpversion>=0.5.3",
+        "pytest-watch>=4.1.0",
+        "tox>=4.0.0",
         "wheel",
         "twine",
         "ipython",
     ],
 }
 
-extras_require['dev'] = (
-    extras_require['dev'] +  # noqa: W504
-    extras_require['test'] +  # noqa: W504
-    extras_require['lint'] +  # noqa: W504
-    extras_require['doc']
+extras_require["dev"] = (
+    extras_require["dev"]
+    + extras_require["test"]
+    + extras_require["lint"]
+    + extras_require["doc"]
 )
 
 
-with open('./README.md') as readme:
+with open("./README.md") as readme:
     long_description = readme.read()
 
 
 setup(
-    name='eth-typing',
+    name="eth-typing",
     # *IMPORTANT*: Don't manually change the version here. Use `make bump`, as described in readme
-    version='3.3.0',
+    version="3.3.0",
     description="""eth-typing: Common type annotations for ethereum python packages""",
     long_description=long_description,
-    long_description_content_type='text/markdown',
-    author='The Ethereum Foundation',
-    author_email='snakecharmers@ethereum.org',
-    url='https://github.com/ethereum/eth-typing',
+    long_description_content_type="text/markdown",
+    author="The Ethereum Foundation",
+    author_email="snakecharmers@ethereum.org",
+    url="https://github.com/ethereum/eth-typing",
     include_package_data=True,
-    python_requires='>=3.7.2, <4',
```

Comment:
This `3.7.2` should stay. We actually don't work with 3.7.0 for a reason I can't remember off the top of my head. 

And just as a general note, we need to be careful changing these because pip validates them on install, and will disallow installation if the version isn't in these bounds, regardless of whether or not a library will actually work on a given version.

Reply:
updating the template so it won't come up again. Thanks!